oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -109,6 +109,40 @@ public class ComputeEngineCredentials extends GoogleCredentials
   static final int MAX_COMPUTE_PING_TRIES = 3;
   static final int COMPUTE_PING_CONNECTION_TIMEOUT_MS = 500;
 
+  public enum GoogleAuthTransport {
```

Comment:
Since GoogleAuthTransport and BindingEnforcement live in ComputeEngineCredentials and not any S2A specific classes, can they have some simple top level javadocs? Ideally if there is public docs that we can link to (this can also be added in the future). 

Would be helpful to answer the combo of when/why users can/ should use them?

Reply:
> can they have some simple top level javadocs?

Done

> Ideally if there is public docs that we can link to (this can also be added in the future).

When the feature (S2A + MTLS + bound tokens) is non-experimental, we will be able to link to public MDS docs. For now, I have marked in the javadocs: "Experimental Feature".

> Would be helpful to answer the combo of when/why users can/ should use them? Also to explain what Google Auth Transport is in this case

Done. we can modify these javadocs again when the feature is non-experimental (public MDS docs are ready).